### DISCUSSION TITLE
Welcome to Math-for-CS-solutions Discussions!

### DISCUSSION THREAD

**Discussion by spamegg1** [spamegg1/Math-for-CS-solutions#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sauerBT** [comment]
Hi @spamegg1, I've just started this course.  I am admittedly new to proofs, but I have university experience through calc 5 as a starting background.  From you chapter 2 proof of sqrt(2)^sqrt(2), the proof you provide doesnt make any sense to me.... especially since it is irrational and you indicate its rational.... any help you can provide?

**Reply by spamegg1** [reply]
That proof trips up a lot of people :smile: 

If you are confused by this course as you go along (it gets MUCH MUCH HARDER) then consider using Susanna Epp's book instead. I have solutions: https://github.com/spamegg1/Epp-Discrete-Math-5th-solutions 

We know $x = \sqrt{2}$ is irrational.

Now we consider the number $y = x^x = (\sqrt{2})^{\sqrt{2}}$.

There are two possible cases: $y$ is rational, or $y$ is irrational. We don't know which. 

In both cases we want to show that there exist two irrational numbers $a,b$ such that $a^b$ is rational.

Case 1. $y$ is rational. Then let $a = x, b = x$. So both $a$ and $b$ are irrational, and $a^b = x^x = y$ is rational.

Case 2. $y$ is irrational. Then let $a = y, b = x$. So both $a$ and $b$ are irrational, and $a^b = y^x = ((\sqrt{2})^{\sqrt{2}})^{\sqrt{2}} = (\sqrt{2})^{\sqrt{2}\sqrt{2}} = (\sqrt{2})^2 = 2$ is rational.

So in both cases we found numbers $a,b$ such that $a$ and $b$ are irrational and $a^b$ is rational.

**Reply by spamegg1** [reply]
The main problem people have here is the difficulty of making assumptions: assumptions that look 1) unjustified, and 2) contradictory; which lead to an argument that feels: 3) circular.

Even MIT students struggle with this. By Prof. Meyer's own admission, 20% of MIT students don't get certain arguments and feel they are circular. He tried to fix it by introducing the proof-by-contradiction-by-the-Well-Ordering-Principle much earlier, but it did not help.

To get better at this, you have to study logic properly from scratch. This course does not do a very good job of teaching logic and reasoning. It assumes you already know, and speeds through everything.

Susanna Epp's book covers logic and reasoning much better. But you have to spend a lot more time.

**Reply by sauerBT** [reply]
Thank you @spamegg1.  Makes perfect sense, and unfortunately my chemical engineering background does not go into logic fundamentals (at all)... I went ahead and purchased Susanna's book to learn from scratch.  The concepts make sense, and I understand your reasoning above.  

Thank you for what you do by the way... Your reviews and solutions have been a life saver.

**Comment by QingMoQian** [comment]
我在完成作业的时候遇到了困难，gpt对我提供的帮助有限，我时常需要怀疑gpt的正确性，感谢你的正确而清晰的论述了我所需要的一切。衷心的感谢！

**Comment by mohammadazeemwani** [comment]
Your recommendation means which of the following?:
1. Do Susanna Epp's book only
2. Do Susanna Epp's book followed my MIT course

**Reply by spamegg1** [reply]
Epp's book by itself is more than sufficient in my opinion. But some people feel like they need to get the "MIT approval" to feel safe and confident in their abilities.

**Reply by mohammadazeemwani** [reply]
Noted and going for Epp's book 🫡
